public/js/wJoystick.js

```diff
@@ -3,25 +3,34 @@ $.widget('custom.JOYSTICK', {
   options: {
     socket: null,
     data: {
-      format: {
-        textColor: 'black',
-        prefix: '',
-        suffix: ''
-      }
-    }
+      format: {}
+    },
+    currentValue: {x:0, y:0}
   },
   _create: function () {
+    const objWidget = this
+    //set a default  data.scale
+    if (!this.options.data.scale) { this.options.data.scale = 1 }
     const objContent = $('<div id="joystick" style="width:200px; height:200px"></div>')
     this.element.children('.widget-content').html(objContent).ready(() => {
       const objJoystick = new JoyStick('joystick', {}, (objData) => {
-        console.log(objData)
+        objWidget.currentValue = objData
+        this._triggerSocketEvent(null, objData)
       })
     })
-  },
-  _setupSocketListener: function () {
-    if (this.options.socket) {
-    } else {
-      console.error('Socket is not defined.')
+    this._triggerSocketEvent(null, { x: 0, y: 0 })
+    // setup the repeater if defined and > 0
+    if (this.options.data.topicPeriodS) {
+      this._repeater = setInterval(() => {  
+        this._triggerSocketEvent(null, objWidget.currentValue )
+      }, this.options.data.topicPeriodS * 1000)
     }
+  },
+  _triggerSocketEvent: function (e, ui) {
```

Comment:
Why does this function have the first argument "e"? It's not used in the function and the two places it's called it's assigned "null".

Reply:
it's the event, I can call it that and since its in the first position I include it.

Reply:
while e for event is common js syntax I went ahead and renamed them to event.